src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -370,6 +370,110 @@ public void It_includes_internals_visible_to()
             AssemblyInfo.Get(assemblyPath)["InternalsVisibleToAttribute"].Should().Be("Tests");
         }
 
+        [Fact]
+        public void Requires_preview_features_enabled()
+        {
+            TestPreviewFeatures(true, true);
+        }
+
+        [Fact]
+        public void Requires_preview_features_disabled()
+        {
+            TestPreviewFeatures(true, false);
+        }
+
+        [Fact]
+        public void Requires_preview_features_not_specified()
+        {
+            TestPreviewFeatures(false, false);
+        }
+
+        private void TestPreviewFeatures(bool addPreviewFeatureProperty, bool enablePreviewFeatures)
+        {
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld", identifier: $"{addPreviewFeatureProperty}_{enablePreviewFeatures}")
+                .WithSource()
+                .WithTargetFramework("net6.0")
+                .WithProjectChanges((path, project) =>
+                {
+                    var ns = project.Root.Name.Namespace;
+
+                    if (addPreviewFeatureProperty)
+                    {
+                        project.Root.Add(
+                            new XElement(ns + "PropertyGroup",
+                                new XElement(ns + "EnablePreviewFeatures", $"{enablePreviewFeatures}")));
+                    }
+                });
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute().Should().Pass();
+
+            var assemblyPath = Path.Combine(buildCommand.GetOutputDirectory("net6.0").FullName, "HelloWorld.dll");
+
+            var parameterlessAttributes = AssemblyInfo.GetParameterlessAttributes(assemblyPath);
+            bool contains = false;
+            foreach (var attribute in parameterlessAttributes)
+            {
+                if (attribute.Equals("RequiresPreviewFeaturesAttribute", System.StringComparison.Ordinal))
+                {
+                    contains = true;
+                    break;
+                }
+            }
+
+            var getValuesCommand = new GetValuesCommand(testAsset, "LangVersion", targetFramework: "net6.0");
```

Comment:
Nit: Don't hardcode this both here and when calling `WithTargetFramework`.  Declare it once as a variable or constant and then use it in both places.